### ISSUE TITLE
HierarchyHandling problematic when directly using layout provider

### ISSUE THREAD

**Issue by uruuru** [eclipse/elk#209]
There's a method in the `RecursiveGraphLayoutEngine` that explicitly computes and sets the `hierarchyHandling` layout option for all hierarchical nodes. This may be problematic when using a layout provider directly. The layout provider cannot really assume that the values have been set and would have to duplicate the logic.

```
private void evaluateHierarchyHandlingInheritance(final ElkNode layoutNode) { ... }
```

**Comment by le-cds** [comment]
We could put that logic into a public utility method which is then also used by the `RecursiveGraphLayoutEngine`?

**Comment by uruuru** [comment]
Who would call it in the case of plain java layout?

**Comment by le-cds** [comment]
Well, if the `RecursiveGraphLayoutEngine` is used the answer is rather obvious. If people call layout algorithms directly I think it's fine to require them to call said method beforehand. First, it's only really necessary if hierarchy is involved. And second, I don't really see why one would call a layout algorithm directly anyhow. And if it is indeed necessary, I guess we may safely assume people who do that to know what they're doing.

**Comment by uruuru** [comment]
Ok.

**Comment by le-cds** [comment]
We should evaluate how this relates to #314.

**Comment by spoenemann** [comment]
It might make sense to move the hierarchy handling computation to LayoutAlgorithmResolver.

**Comment by le-cds** [comment]
Yes, we were thinking along those lines. Since that requires a bit of thought I don't expect that will happen for the next release.
